commands/gitService.go

```diff
@@ -0,0 +1,144 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/go-git/go-git/v5"
+	gitHTTP "github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/section/sectionctl/api"
+)
+
+// GitService interface provides a way to interact with Git
+type GitService interface {
+	UpdateGitViaGit(c *DeployCmd, response UploadResponse) error
+}
+
+// GS ...
+type GS struct{}
+
+// This is far less then ideal, however Kong does not seem to provide a way to inject dependencies into its commands so we must use this for testing
+var globalGitService GitService = &GS{}
```

Comment:
If someone knows another way to do DI with these Kong commands I'd be happy to change this.